src/core/mesh/qgsmeshlayer.cpp

```diff
@@ -352,6 +352,122 @@ QgsMesh3dDataBlock QgsMeshLayer::dataset3dValue( const QgsMeshDatasetIndex &inde
   return block3d;
 }
 
+QgsMeshDatasetValue QgsMeshLayer::dataset1DValue( const QgsMeshDatasetIndex &index, const QgsRectangle &rectangle ) const
+{
+  QgsMeshDatasetValue value;
+  QgsPointXY centerPoint = rectangle.center();
+  const QgsTriangularMesh *mesh = triangularMesh();
+  if ( mesh &&
+       mesh->contains( QgsMesh::Edge ) &&
+       mDataProvider->isValid() &&
+       index.isValid() )
+  {
+    // search for the closest edge in rectangle from point
+    const QList<int> edgeIndexes = mesh->edgeIndexesForRectangle( rectangle );
+    int selectedIndex = -1;
+    double sqrMaxDistFromPoint = pow( rectangle.width() / 2, 2 );
+    QgsPointXY projectedPoint;
+    for ( const int edgeIndex : edgeIndexes )
+    {
+      const QgsMeshEdge &edge = mesh->edges().at( edgeIndex );
+      const QgsMeshVertex &vertex1 = mesh->vertices()[edge.first];
+      const QgsMeshVertex &vertex2 = mesh->vertices()[edge.second];
+      QgsPointXY projPoint;
+      double sqrDist = centerPoint.sqrDistToSegment( vertex1.x(), vertex1.y(), vertex2.x(), vertex2.y(), projPoint );
+      if ( sqrDist < sqrMaxDistFromPoint )
```

Comment:
you should probably look for closest edge, this logic just takes (randomly) a last edge found by index?

Reply:
of course, I've forgot `sqrMaxDistFromPoint=sqrDist;` in the if scope